index.js

```diff
@@ -17,17 +17,102 @@ function simplify (data) {
   return transform(data.value, data.type)
 }
 
-module.exports = {
-  parse: (text) => {
-    try {
-      const parserNE = new nearley.Parser(nearley.Grammar.fromCompiled(grammar))
-      parserNE.feed(text)
-      return parserNE.results[0]
-    } catch (e) {
-      e.message = `Error parsing text '${text}'`
-      throw e
+function stringify (val) {
+  // if (typeof val === 'object' && Array.isArray(val) && val.length === 0) return '[]'
+
+  const { value, type } = val
+  if (type === 'compound') {
+    const str = []
+    const entries = Object.entries(value)
+    for (let i = 0; i < entries.length; i++) {
+      const _type = entries[i][0]
+      let _value = stringify(entries[i][1])
+      if (_type === 'string') _value = normalizeString(_value)
+      str.push(`${_type}:${_value}`)
+    }
+    return `{${str.join(',')}}`
+  } else if (type === 'list') {
+    if (!Array.isArray(value.value)) return '[]'
+    const arrayElements = getArrayValues(value)
+    return `[${arrayElements}]`
+  } else if (type === 'byteArray' || type === 'intArray' || type === 'longArray') {
+    const prefix = getArrayPrefix(type)
+    const arrayElements = getArrayValues(value)
+    return `[${prefix}${arrayElements}]`
+  }
+  // circle back, typed arrays aren't implemented
+  let str = value + getSuffix(value, type)
+  if (type === 'string') str = normalizeString(str)
+  return str
+}
+function normalizeString (str) {
+  if (/,|:|;/g.test(str)) str = `"${str}"`
+  else if (/"/g.test(str)) str = str.replace(/"/, '\"') // eslint-disable-line
```

Comment:
this should be applied before and independently of the previous regex: if it contains quote, it should quoted.

Reply:
done